### PR TITLE
[iso19139:2007] add more generic char/anchor check

### PR DESCRIPTION
**Pull request by pvgenuchten**
Anchor is used in quite places in the inspire profile on iso19139:2007

Add xxx_url property to keep backwards compatibility

Adds tests based on a record, derived from [nationaalgeoregister.nl](https://www.nationaalgeoregister.nl/geonetwork/srv/metadata/f44dac86-2228-412f-8355-e56446ca9933)

### CODE REVIEW

File: owslib/iso.py [hunk 1]
```diff
@@ -27,6 +27,41 @@ def get_namespaces():
 
 namespaces = get_namespaces()
 
+def testFirstCharOrAnchor(md,xpath):
+    """ function which checks for first element both charstring or anchor, if anchor, returns name and uri """
+    r = {'name': '', 'url': ''}
```
Comment (geographika) [inline owslib/iso.py:32]: Just to note docstring refers to URI whereas dict key is URL.

**Comment by geographika** [comment]
@pvgenuchten - do you have a link to the raw XML used as the basis for your test record?

It may also be good to just add the XML snippet that this pull request is solving (with and without a `/gmx:Anchor`). 

The code looks good to me, but I've no experience with `iso19139:2007`. Will leave open for comment for a couple of weeks prior to merging.

**Comment by pvgenuchten** [comment]
updated method text

raw xml is at
https://www.nationaalgeoregister.nl/geonetwork/srv/metadata/f44dac86-2228-412f-8355-e56446ca9933/formatters/xml

**Comment by geographika** [comment]
> updated method text
> 
> raw xml is at https://www.nationaalgeoregister.nl/geonetwork/srv/metadata/f44dac86-2228-412f-8355-e56446ca9933/formatters/xml
> 
> a thing i would like your view on is this: if a list consists of charstring, anchor, charstring, anchor, would you expect:
> 
> ```
> {
> name: [foo,la,lo,li]
> url: [http://la,http:li]
> }
> ```
> 
> or
> 
> ```
> {
> name: [foo,la,lo,li]
> url: [None,http://la,None,http:li]
> }
> ```


I'd imagine it would be expected the name and URL list lengths would match, although not sure about use of `None`. An empty string may be better in this case?

@pvgenuchten do you have an example of the above from the XML you linked to that I could test the above question with?
I've tested the branch and new properties with the following Python:

```python
>>> from owslib.iso import *
>>>
>>> md = MD_Metadata(etree.parse('test.xml'))
>>> iden = md.identification[0]
>>> iden.contact[0].organization_url
'http://standaarden.overheid.nl/owms/terms/Ministerie_van_Defensie'
>>> iden.contact[0].name_url
''
>>> iden.contact[0].name
'Ministerie van Defensie, Koninklijke Marine, Dienst der Hydrografie'
>>> iden.otherconstraints
['Geen beperkingen', 'Er zijn geen condities voor toegang en gebruik', 'Geen beperkingen voor publieke toegang']

>>> dist = md.distribution
>>> dist.specification_url
'http://inspire.ec.europa.eu/id/document/tg/hy'
```

**Comment by geographika** [comment]
Tested this branch with the live URL. 

```python
from owslib.util import openURL
from owslib.iso import MD_Metadata
from lxml import etree
url = 'https://www.nationaalgeoregister.nl/geonetwork/srv/api/records/1e9cf0ef-f905-46df-b4f8-2bab7a3808bb/formatters/xml'

response = openURL(url)
xml_content = response.read()
xml_tree = etree.fromstring(xml_content)
md = MD_Metadata(xml_tree)


iden = md.identification[0]
# ['Te gebruiken onder voorwaarden van ESRI licentie', 'Niet Commercieel, Geen Afgeleide Werken, Naamsvermelding verplicht, organisatienaam']
```

This picks up values from both `CharacterString` and `Anchor`. 

```xml
<gmd:otherConstraints>
<gco:CharacterString>Te gebruiken onder voorwaarden van ESRI licentie</gco:CharacterString>
</gmd:otherConstraints>
...
<gmd:otherConstraints>
<gmx:Anchor xlink:href="http://creativecommons.org/licenses/by-nc-nd/4.0/deed.nl">
Niet Commercieel, Geen Afgeleide Werken, Naamsvermelding verplicht, organisatienaam
</gmx:Anchor>
</gmd:otherConstraints>
```

Failing CI issues are unrelated. Will merge shortly unless any issues raised.

**Pull request merged**